contents/tutorials/test-frontend-feature-flags.md

```diff
@@ -14,32 +14,80 @@ Combining both testing and feature flags can be a bit tricky. Tests generally ch
 
 To do this, you need to mock the flags to access the other variations. This tutorial shows you how to do that by creating a React app with Jest tests, adding PostHog, then setting up tests that work with feature flags by mocking PostHog.
 
-## Creating a React app
+## Creating a Vite-React app and setup Vitest
 
-First, create a basic React app. Make sure [Node is installed](https://nodejs.dev/en/learn/how-to-install-nodejs/) then run `npx create-react-app` with the name of your app. We named ours `flag-test`.
+First, ensure [Node.js is installed](https://nodejs.dev/en/learn/how-to-install-nodejs/) (version 18.0 or newer). Then create a new React app with Vite: We named ours `flag-test`.
 
 ```bash
-npx create-react-app flag-test
+npm create vite@latest flag-test -- --template react
 ```
 
-After creating the app, in the newly created `flag-test` folder, a test is automatically created in `src/App.test.js` that looks like this:
+After creating the app, in the newly created `flag-test` folder, but by default the test framework isn't installed at the project yet, so let's add it.
+
+```bash
+cd flag-test
+npm add -D vitest @testing-library/react @testing-library/jest-dom
+```
+
+Since we're about to test `components` directly, we must create a file called `vitest.setup.js` at our project root importing the required dependency:
```

Comment:
```suggestion
Since we're about to test `components` directly, we must create a file called `vitest.setup.js` at our project root and import the required dependency:
```